### ISSUE TITLE
Question about cbDice Training Time

### ISSUE THREAD

**Issue by PengchengShi1220** [jmlipman/TopoMortar#2]
Hello, thanks for your outstanding evaluation work! I have a question about the training time of cbDice. Its computation time is affected by distance map calculations and skeletonization methods. In your paper and the [TopoMortar README](https://github.com/jmlipman/TopoMortar?tab=readme-ov-file#latest-results-last-update-05-mar-25), I noticed that cbDice’s training time is significantly longer than clDice’s.  

Based on my previous records in the [cbDice Time Comparisons](https://github.com/PengchengShi1220/cbDice?tab=readme-ov-file#time-comparisons):  
- With Topology-Preserving Skeletonization, CE+Dice+cbDice takes 4.6 times longer than CE+Dice.
- With Morphological Skeletonization (iter_=10), CE+Dice+cbDice takes 1.3 times longer than CE+Dice.

When the same skeletonization method is applied, the training times of CE+Dice+cbDice and CE+Dice+clDice are nearly identical. The [cbDice Installation Guide](https://github.com/PengchengShi1220/cbDice?tab=readme-ov-file#installation-guide) recommends installing cuCIM and CuPy for GPU-accelerated distance transforms:  

```
# For CUDA 12.x
pip install monai cucim-cu12 cupy==12.3  

# For CUDA 11.x
pip install monai cucim-cu11 cupy-cuda11x  
```

Have you utilized cuCIM and CuPy for GPU acceleration? If not, I recommend doing so, as it significantly reduces computation time. 

Additionally, I suggest comparing clDice and cbDice using the same skeletonization method to ensure consistency, as the choice of skeletonization impacts both training time and results.

**Comment by jmlipman** [comment]
Hi,

Thanks for the interest and input! 

#### 1) Skeletonization
The skeletonization algorithm totally makes a difference. This is true for all methods. Even with clDice I don't get great skeletons; each image (even each image region) has a different optimal number of iterations. For the comparison done is this work, I did not tune extensively the hyper-parameters. Instead, I mostly took the code that was shared and ensured that it was doing what the paper says.

#### 2) Training time measurement
I did not use cuCIM nor CuPy. I suppose this would also benefit other loss functions like Warping loss. I will add to the TODO list adding a new column to the table reporting the time with these libraries installed. Thank you so much for the suggestion.

I run with the current setup a quick comparison between CEDice, clDice and cbDice.

![Image](https://github.com/user-attachments/assets/dec080ea-987c-4c55-99c6-66b574489f4b)

CEDice took about 1.23 seconds and clDice 12.55 seconds. For cbDice, I had to reduce the number of iterations from 1000 to 10, and it took about 25 seconds.

I will leave this issue open as a reminder to re-measure the training time with those libraries installed.

**Comment by PengchengShi1220** [comment]
Hi,

Thank you for your prompt and detailed reply. I appreciate that you’ll include adding a new column to the table reporting the time with cuCIM and CuPy libraries installed. Looking forward to the updates!
